File: info/1. Алгоритмы/5. Вычислительные алгоритмы/5. Ряд Фибоначчи/code sample/sample1/Sample1.java
package sample1;

import java.math.BigInteger;

public class Sample1 {

    public static void main(String[] args) {
        for (int i = 0; i <= 10; i++) {
            System.out.println(i + " -> " + fibonacciSequince(i));
        }
    }

    public static BigInteger fibonacciSequince(int n) {
        BigInteger result = BigInteger.ZERO;
        BigInteger next = BigInteger.ONE;
        int currentIndex = 0;
        for (; currentIndex < n;) {
            next = result.add(next);
            result = next.subtract(result);
            currentIndex++;
        }
        return result;
    }

}

File: info/1. Алгоритмы/5. Вычислительные алгоритмы/5. Ряд Фибоначчи/code sample/sample_1.py

def fibonacci_sequince(n):
    result = 0
    next_element = 1
    index = 0
    while index < n:
        next_element, result = next_element+result, next_element
        index += 1
    return result


for i in range(10):
    print(i, " -> ", fibonacci_sequince(i))
